### PR TITLE
Training scripts

### PR DESCRIPTION
**Pull request by ishayrao**
Added train.py and model configs and python scripts, along with transforms.py.

### CODE REVIEW

**Comment by ishayrao** [comment]
would look at train.py, faster_rcnn_detector.py, ssdlite_detector.py, transforms.py and whichever configs are relevant. model_summaries outputs the model architectures that i put in the doc.

File: coco_models/src/train.py [hunk 1]
```diff
@@ -4,80 +4,250 @@
...
+        # Training phase
+        model.train()
+        train_loss = 0.0
+    
+        train_pbar = tqdm(train_loader, desc=f'Epoch {epoch+1}/{cfg.training.num_epochs} [Train]')
+        for batch_idx, (data, targets) in enumerate(train_pbar):
+            # Move data to device
+            data = data.to(device)
+            # move target tensors to device as list of dicts 
+            targets_on_device = [{k: v.to(device) for k, v in t.items()} for t in targets] #standard format for all models
```
Comment (Rob1in) [inline coco_models/src/train.py:42]: my favorite for loop. 

1. Option 1: the whole dataset fits in GPU memory -> we set pin_memory=True 
2. Option 2: modify to collate function to parallelize the processing.

We will talk about that.

File: coco_models/src/train.py [hunk 2]
```diff
@@ -4,80 +4,250 @@
...
+            
+            optimizer.zero_grad()
+
+            if cfg.model.name in ["faster_rcnn", "ssdlite"]: #for torchvision models: they return a Dict[Tensor] which contains classification and regression losses
+                loss_dict = model(data, targets_on_device)
+                loss = sum(loss_dict.values()) #sum of classification and regression losses
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply coco_models/src/train.py:48]: This is important also for orders of magnitude, if the classification loss is huge compared to the bbox one, the focus of the optimizer will be on the classification loss. Might be some good  scheduling strategy for the weights of the losses
